src/wheel/bdist_wheel.py

```diff
@@ -432,38 +431,20 @@ def _ensure_relative(self, path):
 
     @property
     def license_paths(self):
-        metadata = self.distribution.get_option_dict("metadata")
+        metadata = self.distribution.metadata
         files = set()
-        patterns = sorted(
-            {option for option in metadata.get("license_files", ("", ""))[1].split()}
-        )
 
-        if "license_file" in metadata:
+        license_file = getattr(metadata, "license_file", None)
+        if license_file:
             warnings.warn(
                 'The "license_file" option is deprecated. Use '
                 '"license_files" instead.',
                 DeprecationWarning,
             )
-            files.add(metadata["license_file"][1])
-
-        if "license_file" not in metadata and "license_files" not in metadata:
-            patterns = ("LICEN[CS]E*", "COPYING*", "NOTICE*", "AUTHORS*")
-
-        for pattern in patterns:
-            for path in iglob(pattern):
-                if path.endswith("~"):
-                    log.debug(
-                        f'ignoring license file "{path}" as it looks like a ' f"backup"
-                    )
-                    continue
-
-                if path not in files and os.path.isfile(path):
-                    log.info(
-                        f'adding license file "{path}" (matched pattern "{pattern}")'
-                    )
-                    files.add(path)
-
-        return files
+            files.add(license_file)
+
+        license_files = getattr(metadata, "license_files", None) or []
```

Comment:
Is the attribute entirely absent if the option hasn't been provided?

Reply:
I believe the attribute is always present if `setuptools` is used.
Is there any chance `bdist_wheel` runs with `distutils`?

Reply:
> I believe the attribute is always present if `setuptools` is used. Is there any chance `bdist_wheel` runs with `distutils`?

`bdist_wheel` is a setuptools extension so I don't think this could ever happen.

Reply:
Ok, so I will remove the `getattr` and use the attribute directly. Thanks Alex.

Reply:
Done in https://github.com/pypa/wheel/pull/466/commits/7545b8a6c849c039587f40b75817292dbcbf3045.